PGM/src/main/java/tc/oc/pgm/menu/gui/SettingMenuHelper.java

```diff
@@ -0,0 +1,49 @@
+package tc.oc.pgm.menu.gui;
+
+import me.anxuiz.settings.Setting;
+import me.anxuiz.settings.bukkit.PlayerSettings;
+import tc.oc.commons.bukkit.broadcast.BroadcastSettings;
+import tc.oc.commons.bukkit.punishment.PunishmentMessageSetting;
+import tc.oc.commons.bukkit.users.JoinMessageSetting;
+import tc.oc.commons.bukkit.whisper.WhisperSettings;
+import tc.oc.pgm.damage.DamageSettings;
+import tc.oc.pgm.death.DeathMessageSetting;
+import tc.oc.pgm.picker.PickerSettings;
+import tc.oc.pgm.playerstats.StatSettings;
+import tc.oc.pgm.settings.ObserverSetting;
+import tc.oc.pgm.settings.Settings;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class SettingMenuHelper {
+
+    private static Map<Setting, SettingType> settings = new HashMap<>();
+
+    public static void initializeSettings() {
+        Setting[] chat = {DeathMessageSetting.get(), JoinMessageSetting.get(), WhisperSettings.receive(), PunishmentMessageSetting.get(), BroadcastSettings.TIPS};
```

Comment:
This should be done by modifying the `Setting` interface. I would add a key-value store to `Setting` in the form of `Setting#getExtra() -> Map<String, String>`.